### PR TITLE
Get it

### PR DESCRIPTION
**Pull request by IvanAleinikau**

### CODE REVIEW

File: .idea/libraries/Dart_Packages.xml [hunk 1]
```diff
@@ -415,6 +415,13 @@
             </list>
           </value>
         </entry>
+        <entry key="get_it">
+          <value>
+            <list>
+              <option value="$USER_HOME$/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/get_it-7.2.0/lib" />
```
Comment (Platton96) [inline .idea/libraries/Dart_Packages.xml:421]: congig gitignore

File: lib/core/bloc/bloc_auth/auth_bloc.dart [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import 'package:shop/core/bloc/bloc_auth/auth_event.dart';
 import 'package:shop/core/bloc/bloc_auth/auth_state.dart';
 import 'package:shop/core/bloc/bloc_splash/splash_bloc.dart';
 import 'package:shop/core/bloc/bloc_splash/splash_event.dart';
 import 'package:shop/data/repositories/auth_repository.dart';
 
 class AuthBloc extends Bloc<AuthEvent, AuthState> {
-  AuthRepository repository = AuthRepository();
-
+  var repository = GetIt.instance<AuthRepository>();
```
Comment (Platton96) [inline lib/core/bloc/bloc_auth/auth_bloc.dart:12]: use final

File: lib/core/bloc/bloc_fqa/fqa_bloc.dart [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:get_it/get_it.dart';
 import 'package:shop/core/bloc/bloc_fqa/fqa_event.dart';
 import 'package:shop/core/bloc/bloc_fqa/fqa_state.dart';
 import 'package:shop/core/models/question_answer_model.dart';
 import 'package:shop/data/repositories/local_database_repository.dart';
 
 class FqaBloc extends Bloc<FqaEvent, FqaState> {
-  LocalDatabaseRepository database = LocalDatabaseRepository();
+  var repository = GetIt.instance<LocalDatabaseRepository>();
```
Comment (Platton96) [inline lib/core/bloc/bloc_fqa/fqa_bloc.dart:11]: final

File: lib/core/bloc/bloc_register/register_bloc.dart [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+import 'package:get_it/get_it.dart';
 import 'package:shop/core/bloc/bloc_register/register_event.dart';
 import 'package:shop/core/bloc/bloc_register/register_state.dart';
 import 'package:shop/core/bloc/bloc_splash/splash_bloc.dart';
 import 'package:shop/core/bloc/bloc_splash/splash_event.dart';
 import 'package:shop/data/repositories/auth_repository.dart';
 
 class RegisterBloc extends Bloc<RegisterEvent, RegisterState> {
-  AuthRepository repository = AuthRepository();
+  var repository = GetIt.instance<AuthRepository>();
```
Comment (Platton96) [inline lib/core/bloc/bloc_register/register_bloc.dart:12]: check all plases

File: lib/main.dart [hunk 5]
```diff
@@ -41,47 +42,48 @@ class _AppState extends State<App> {
...
-          ),
-          BlocProvider<ShoppingCartBloc>(
-            create: (context) => ShoppingCartBloc(),
-          ),
-          BlocProvider<SplashBloc>(
-            create: (context) => splashBloc..add(CheckCurrentUser()),
-          ),
+      providers: [
+        BlocProvider<SavedNewsBloc>(
+          create: (context) => SavedNewsBloc(),
```
Comment (Platton96) [inline lib/main.dart:47]: bad solution, we should wrap page to blocProvider, if bloc will be used on few peges then register in this place

File: lib/core/bloc/bloc_auth/auth_bloc.dart (hunk 2 above)
Comment (IvanAleinikau) [reply lib/core/bloc/bloc_auth/auth_bloc.dart:12]: done

File: lib/core/bloc/bloc_fqa/fqa_bloc.dart (hunk 3 above)
Comment (IvanAleinikau) [reply lib/core/bloc/bloc_fqa/fqa_bloc.dart:11]: done

File: lib/core/bloc/bloc_register/register_bloc.dart (hunk 4 above)
Comment (IvanAleinikau) [reply lib/core/bloc/bloc_register/register_bloc.dart:12]: done

File: lib/main.dart (hunk 5 above)
Comment (IvanAleinikau) [reply lib/main.dart:47]: done

Comment (IvanAleinikau) [reply lib/main.dart:47]: done

**Pull request merged**